TipCatDotNet.Api/Services/HospitalityFacilities/StripeAccountService.cs

```diff
@@ -0,0 +1,150 @@
+                return account.Id;
+            }
+            catch (StripeException ex)
+            {
+                return Result.Failure<string>(ex.Message);
+            }
+        }
+
+
+        public async Task<Result> Update(MemberRequest request, CancellationToken cancellationToken)
+        {
+            var member = await _context.Members
+                .Where(m => m.Id == request.Id)
+                .SingleOrDefaultAsync(cancellationToken);
+
+            var (_, isFailure, isMatch, error) = await AccountsMatches(member!.Id, member!.StripeAccountId, cancellationToken);
+
+            if (isFailure)
+                return Result.Failure(error);
+
+
+            var updateOptions = new AccountUpdateOptions
+            {
+                Individual = new AccountIndividualOptions
+                {
+                    FirstName = request.FirstName,
+                    LastName = request.LastName,
+                    Email = request.Email
+                }
+            };
+            try
+            {
+                var updatedAccount = await _accountService.UpdateAsync(member!.StripeAccountId, updateOptions, cancellationToken: cancellationToken);
+                return Result.Success();
+            }
+            catch (StripeException ex)
+            {
+                return Result.Failure<PaymentIntent>(ex.Message);
+            }
+        }
+
+
+        /// <summary>
+        /// Remove stripe account.
+        /// Accounts created using test-mode keys can be deleted at any time.
+        /// Custom or Express accounts created using live-mode keys can only be deleted once all balances are zero.
+        /// </summary>
+        public async Task<Result> Remove(int memberId, CancellationToken cancellationToken)
+        {
+            var member = await _context.Members
+                .Where(m => m.Id == memberId)
+                .SingleOrDefaultAsync(cancellationToken);
+
+            var (_, isFailure, isMatch, error) = await AccountsMatches(member!.Id, member!.StripeAccountId, cancellationToken);
+
+            if (isFailure)
+                return Result.Failure(error);
+
+
+            try
+            {
+                var updatedAccount = await _accountService.DeleteAsync(member!.StripeAccountId, cancellationToken: cancellationToken);
+                return Result.Success();
+            }
+            catch (StripeException ex)
+            {
+                return Result.Failure<PaymentIntent>(ex.Message);
+            }
+        }
+
+
+        private async Task<Result<bool>> AccountsMatches(int memberId, string accountId, CancellationToken cancellationToken)
+        {
+            var isMatch = true;
+            try
+            {
+                var account = await _accountService.GetAsync(accountId, cancellationToken: cancellationToken);
+                if (account.Metadata["MemberId"] == memberId.ToString())
```

Comment:
Why don't just get it from a table?

Reply:
In this case I create this method to be ensure that is it true that updated account in stripe strictly related with ourMember. In case somehow stripeAccountId in our db will be changed.

Reply:
Ok